tests/integration_tests/modules/test_lxd.py

```diff
@@ -62,27 +137,81 @@ def validate_storage(validate_client, pkg_name, command):
     return log
 
 
+def validate_preseed_profiles(client, preseed_cfg):
+    for src_profile in preseed_cfg["profiles"]:
+        profile = yaml.safe_load(
+            client.execute(f"lxc profile show {src_profile['name']}")
+        )
+        assert src_profile["config"] == profile["config"]
+
+
+def validate_preseed_storage_pools(client, preseed_cfg):
+    for src_storage in preseed_cfg["storage_pools"]:
+        storage_pool = yaml.safe_load(
+            client.execute(f"lxc storage show {src_storage['name']}")
+        )
+        assert storage_pool["config"]["source"] == storage_pool["config"].pop(
+            "volatile.initial_source"
+        )
+        assert storage_pool["config"] == src_storage["config"]
+        assert storage_pool["driver"] == src_storage["driver"]
+
+
+def validate_preseed_projects(client, preseed_cfg):
+    for src_project in preseed_cfg["projects"]:
+        project = yaml.safe_load(
+            client.execute(f"lxc project show {src_project['name']}")
+        )
+        assert [
+            "/1.0/profiles/bionic-vm-lxc-setup",
+            "/1.0/profiles/default",
+            "/1.0/networks/lxdbr0",
+        ] == project.pop("used_by")
+        assert project == src_project
+
+
 @pytest.mark.no_container
 @pytest.mark.user_data(STORAGE_USER_DATA.format("btrfs"))
 def test_storage_btrfs(client):
     validate_storage(client, "btrfs-progs", "mkfs.btrfs")
 
 
 @pytest.mark.no_container
-@pytest.mark.user_data(STORAGE_USER_DATA.format("lvm"))
-def test_storage_lvm(client):
-    log = client.read_from_file("/var/log/cloud-init.log")
-
-    # Note to self
-    if (
-        "doesn't use thinpool by default on Ubuntu due to LP" not in log
-        and "-kvm" not in client.execute("uname -r")
-    ):
-        warnings.warn("LP 1982780 has been fixed, update to allow thinpools")
-    validate_storage(client, "lvm2", "lvcreate")
-
-
-@pytest.mark.no_container
-@pytest.mark.user_data(STORAGE_USER_DATA.format("zfs"))
-def test_storage_zfs(client):
-    validate_storage(client, "zfsutils-linux", "zpool")
+@pytest.mark.user_data(STORAGE_PRESEED_USER_DATA.format("btrfs"))
+def test_storage_preseed_btrfs(client):
+    validate_storage(client, "btrfs-progs", "mkfs.btrfs")
+    src_cfg = yaml.safe_load(STORAGE_PRESEED_USER_DATA.format("btrfs"))
+    preseed_cfg = src_cfg["lxd"]["preseed"]
+    validate_preseed_profiles(client, preseed_cfg)
+    validate_preseed_storage_pools(client, preseed_cfg)
+    validate_preseed_projects(client, preseed_cfg)
+
+    @pytest.mark.no_container
```

Comment:
The rest of the tests in the file aren't actually getting run because of the indentation here.